### ISSUE TITLE
Shift + control + / on macOS is faulty

### ISSUE THREAD

**Issue by bsolomon-bw** [wez/wezterm#1907]
### What Operating System(s) are you seeing this problem on?

macOS

### WezTerm version

20220421-183827-26500c67

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

`Control + shift + \<key\>` doesn't seem to work properly (macOS 12.2.1) -- I just get the key without shift or control.

This is important because in Emacs, shift often reverses whatever operation you're performing, and most operations use control.  For example in undo-tree mode, `control-/` is undo; `shift-control-/` is redo.

I suspect this is because wezterm is capturing control+shift for tab switching -- can this be made configurable to a different hotkey?

### Configuration

```lua
local wz = require 'wezterm'

return {
  font = wz.font("Go Mono"),
  font_size = 15,
  color_scheme = "Tomorrow",

  use_fancy_tab_bar = false,
  hide_tab_bar_if_only_one_tab = true,
}
```

### Expected Behavior

I expect my terminal to capture (and forward, if I'm in SSH) combinations with control+shift, with a custom config if necessary.

I actually found this terminal because none of the other fast modern terminals I found (alacritty, kitty) support meta forwarding over SSH well on macOS, so it's a meaningful usecase for Wezterm :)

### Anything else?

Great terminal so far!  I love the Lua configs.  Everything should offer Lua as a config language.

**Comment by wez** [comment]
This should be fixed now in `main`.

It typically takes about an hour before fixes are available as nightly builds for all platforms.  Linux builds are the fastest to build and are often available within about 20 minutes.  Windows and macOS builds take a bit longer.

Please take a few moments to try out the fix and let me know how that works out.

If you prefer to use packages provided by your distribution or package manager of choice and don't want to replace that with a nightly download, keep in mind that you can download portable packages (eg: a `.dmg` file on macOS, a `.zip` file on Windows and an `.AppImage` file on Linux) that can be run without permanently installing or replacing an existing package, and can then simply be deleted once you no longer need them.

If you are eager and can [build from source](https://wezfurlong.org/wezterm/install/source.html) then you may be able to try this out more quickly.

**Comment by bsolomon-bw** [comment]
@wez -- wow, thanks for the quick turnaround!  unfortunately, this just seems to produce `shift+/`...  😅 wish I understood how the macOS modifier key stuff worked so I could be more useful here!  For what it's worth, it doesn't seem to work in any other terminal I have either....  but it does work in the Emacs GUI on macOS.  Hmmmmm....

**Comment by wez** [comment]
Hmmm, yeah, there was a problem with that fix; I took another run at it in bd70cad6fa89a1dbcf0f8d6f7826e31d319d0e9b
Now wezterm matches the output that I see on linux (ssh'ing into a linux box from the mac, and running directly on the linux box itself) for `showkey -a` when pressing `ctrl-/` and `shift-ctrl-/`.

Note that even with that, if emacs doesn't respect that key press for the other terminal emulators either, it may indicate a problem with emacs!

**Comment by bsolomon-bw** [comment]
@wez rebuilt -- I think I might be encountering the issue described here: https://stackoverflow.com/a/32036547

FWIW, these are the keys I get from `showkey -a` on the remote host, for `control+/` (shown here as `^_`) and `control+shift+/` (shown here as `^?`)

```
^_       31 0037 0x1f
^?      127 0177 0x7f
```

--- 

In Emacs, when I run `C-h k` (to show the key combination pressed next), `control+/` shows the expected result:

```
C-_ runs the command undo (found in global-map), which is an
interactive native compiled Lisp function in
‘/snap/emacs/1490/usr/share/emacs/28.1/lisp/simple.el’.

It is bound to C-_, <undo>, C-/, C-x u.
```

but `control+shift+/` simply shows DEL.

Curious....

**Comment by wez** [comment]
Yeah, `ctrl-shift-/` produces `0x7f` which is `DEL`.  The standard terminal keyboard encoding doesn't have a way to represent that key combination.

You have a couple of options on how to proceed:

* Redefine `ctrl-shift-/` to send an alternative sequence of your own devising using [SendString](https://wezfurlong.org/wezterm/config/lua/keyassignment/SendString.html), and then teach emacs to recognize that accordingly
* If emacs supports CSI-u encoding, or you don't fancy inventing your own sequence, you may wish to try [enable_csi_u_key_encoding](https://wezfurlong.org/wezterm/config/lua/config/enable_csi_u_key_encoding.html) and/or teach emacs to recognize the unambiguous csi-u form for `ctrl-shift-/`
